lib/spack/spack/environment/environment.py

```diff
@@ -769,6 +781,28 @@ def _create_environment(path):
     return Environment(path)
 
 
+def config_stage_dir(manifest_dir: Union[str, pathlib.Path]) -> str:
+    """Path to locally staged environment configuration files.
+
+    Args:
+        manifest_dir:  directory containing the environment manifest file
+
+    Returns:  directory for any staged environment configuration files.
+    """
+    return os.path.join(env_subdir_path(manifest_dir), "config")
+
+
+def env_subdir_path(manifest_dir: Union[str, pathlib.Path]) -> str:
```

Comment:
It seems like we should be able to be defined in fewer places, do we need all three entrypoints?

Reply:
At one point in my investigations having some of these new module-level routines made sense but, since I wasn't sure if it would pass the review, I kept the existing uses associated with the environment.  I'll revisit getting rid of the extras.

Reply:
Done (I think).